### PR TITLE
Cleanup and Py3 Support

### PR DESCRIPTION
**Pull request by iamjem**
This just does some general cleanup, and adds explicit testing with tox for various python 3 versions.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -11,24 +9,20 @@
 
 
 requires = [
-    'requests',
+    'requests>=2.6.0',
 ]
 
-version = ''
-with open('telex/__init__.py', 'r') as fd:
```
Comment (sigmavirus24) [inline setup.py:18]: I'm confused, what does removing this have to do with Python 3?

File: setup.py [hunk 2]
```diff
@@ -42,7 +36,10 @@
         'License :: OSI Approved :: MIT License',
         'Natural Language :: English',
         'Programming Language :: Python',
-        "Programming Language :: Python :: 2",
         'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3.3',
```
Comment (sigmavirus24) [inline setup.py]: I'd rather we just support 3.6. 3.3 support is being dropped by Requests so at least make the lower limit 3.4 if we want to support such a large swath.

File: telex/__init__.py [hunk 3]
```diff
@@ -1,4 +1 @@
-# -*- coding: utf-8 -*-
-
-
 __version__ = '0.0.1'
```
Comment (sigmavirus24) [inline telex/__init__.py]: If we're statically declaring the version in `setup.py` why not remove this?

File: tests/requirements.txt [hunk 4]
```diff
@@ -1,4 +0,0 @@
-httmock==1.2.3
```
Comment (sigmavirus24) [inline tests/requirements.txt:1]: Why not keep this and alter the `tox.ini` to be smarter?

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[tox]
+envlist = py27,py33,py34,py35,py36
+
+[testenv]
+deps=pytest
```
Comment (sigmavirus24) [inline tox.ini]: We could do `-rtests/requirements.txt` instead.

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,7 @@
+telex.egg-info*
+*/telex.egg-info/*
+build/*
+dist/*
 *.pyc
-.venv/
```
Comment (sigmavirus24) [inline .gitignore]: I don't think it hurts to keep these two around. I get the sense that the team doesn't necessary embrace `tox` so having virtualenv directories be ignored seems like a harmless idea.

File: setup.py [hunk 7]
```diff
@@ -11,24 +9,20 @@
...
-
+with open('README.md') as f:
+    readme = f.read()
+with open('LICENSE') as f:
+    license = f.read()
 
 setup(
     name='telex',
-    version=version,
+    version='0.0.1',
```
Comment (sigmavirus24) [inline setup.py]: Also, a revision like this probably warrants a new version number.

File: setup.py [hunk 8]
```diff
@@ -11,24 +9,20 @@
...
-with open('telex/__init__.py', 'r') as fd:
-    version = re.search(r'^__version__\s*=\s*[\'"]([^\'"]*)[\'"]',
-                        fd.read(), re.MULTILINE).group(1)
-
-with open('README.md', 'r', 'utf-8') as fd:
-    readme = fd.read()
-
+with open('README.md') as f:
+    readme = f.read()
+with open('LICENSE') as f:
```
Comment (sigmavirus24) [inline setup.py]: We don't want to read the text of the license into memory and shove it into our package metadata. That will break rendering of the README if we upload to PyPI.

File: setup.py [hunk 9]
```diff
@@ -11,24 +9,20 @@
...
+    license = f.read()
 
 setup(
     name='telex',
-    version=version,
+    version='0.0.1',
     description='Python telex client.',
     long_description=readme,
-    license='MIT',
+    license=license,
```
Comment (sigmavirus24) [inline setup.py]: As I mentioned above, this should stay as `MIT`. Otherwise, PyPI will be fail to render the information about the package.

**Review by sigmavirus24 — CHANGES_REQUESTED** [review]
The substance of this change is fine with the exception of the license change in `setup.py`. I have some suggestions and concerns around the rest of this.

**Comment by sigmavirus24** [comment]
We should probably turn on Travis or Circle for this repository too

File: setup.py (hunk 1 above)
Comment (iamjem) [reply setup.py:18]: Nothing, it just seemed unnecessary. Could also import `__version__` explicitly I guess too.

File: setup.py (hunk 9 above)
Comment (iamjem) [reply setup.py]: Hrmm we do this in some of our other packages too. I guess we should fix that as we're going through them for this if applicable.

File: tox.ini (hunk 5 above)
Comment (iamjem) [reply tox.ini]: Right I did this on another package I'm fixing right now. Wasn't aware it was an option till I was looking through the docs more. I can update here I guess, ultimately there's so few test requirements I don't think it matters one way or another.

File: .gitignore (hunk 6 above)
Comment (iamjem) [reply .gitignore]: The team has largely switched to using virtualenvwrapper and we typically keep the envs outside of the project (we use to all keep them in `.venv`). But, still valid point, might as well leave it.

**Comment by iamjem** [comment]
Thanks for the great feedback. I believe I answered all of your comments, let me know if there's anything else I should consider changing.

**Review by sigmavirus24 — APPROVED** [review]
Looks good! :shipit:

**Pull request merged**